src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OverrideEqualsOnOverloadingOperatorEquals.Fixer.cs

```diff
@@ -27,10 +23,57 @@ public sealed override FixAllProvider GetFixAllProvider()
         }
 
         public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
-        {                              
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await new Task(() => { });
-            throw new NotImplementedException();
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            var typeDecl = root.FindNode(context.Span);
+            typeDecl = SyntaxGenerator.GetGenerator(context.Document).GetDeclaration(typeDecl);
+            if (typeDecl == null)
+            {
+                return;
+            }
+
+            // CONSIDER: Do we need to confirm that System.Object.Equals isn't shadowed in a base type?
+
+            // We cannot have multiple overlapping diagnostics of this id.
+            var diagnostic = context.Diagnostics.Single();
+
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    MicrosoftApiDesignGuidelinesAnalyzersResources.OverrideEqualsOnOverloadingOperatorEqualsCodeActionTitle,
+                    cancellationToken => OverrideObjectEquals(context.Document, typeDecl, cancellationToken)),
```

Comment:
I believe it defaults to the title.  Or are you suggesting that this particular fixer needs to use a different value?